services/provider/server/server.go

```diff
@@ -687,6 +694,15 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			storageClusters, err := util.GetStorageClustersInNamespace(ctx, s.client, s.namespace)
+			if err != nil || len(storageClusters.Items) == 0 {
+				return nil, status.Errorf(codes.Internal, "failed to get storage cluster %v", err)
+			}
+			kernelMountOptions := util.GetCephFSKernelMountOptions(&storageClusters.Items[0])
+			if kernelMountOptions == kernelMountOptionSecure {
+				cephfsStorageClassData[kernelMountOptionsKey] = kernelMountOptionSecure
```

Comment:
We should not piggyback csi config info onto the storage class resource. 
CSI config needs to be returned as its own resource